src/GafferScene/IECoreScenePreview/PrimitiveAlgo.cpp

```diff
@@ -0,0 +1,1188 @@
+			// value, so we don't throw an exception if they don't match, we just emit a warning, and use the
+			// default value.
+			if( resultTypeId == IECoreScene::MeshPrimitiveTypeId )
+			{
+				const MeshPrimitive *sourceMesh = static_cast< const MeshPrimitive *>( primitives[i].first );
+				const std::string interp = sourceMesh->interpolation();
+				if(
+					resultMeshInterpolation != "" &&
+					interp != resultMeshInterpolation
+				)
+				{
+					msg( Msg::Warning, "mergePrimitives",
+						fmt::format(
+							"Ignoring mismatch between mesh interpolations {} and {} and defaulting to linear",
+							resultMeshInterpolation, interp
+						)
+					);
+					resultMeshInterpolation = "";
+				}
+			}
+			else if( resultTypeId == IECoreScene::CurvesPrimitiveTypeId )
+			{
+				const CurvesPrimitive *sourceCurves = static_cast< const CurvesPrimitive *>( primitives[i].first );
+				CubicBasisf basis = sourceCurves->basis();
+				bool periodic = sourceCurves->periodic();
+				if( periodic != resultCurvesPeriodic )
+				{
+					throw IECore::Exception( "Cannot merge periodic and non-periodic curves" );
+				}
+
+				if(
+					resultCurvesBasis != invalidBasis &&
+					basis != resultCurvesBasis
+				)
+				{
+					msg( Msg::Warning, "mergePrimitives",
+						"Ignoring mismatch in curve basis and defaulting to linear"
+					);
+					resultCurvesBasis = invalidBasis;
+				}
+			}
+		}
+
+		// Process all the primvars for this primitive, adding new entries to the varInfo list, or
+		// checking that existing entries match correctly
+		for( const auto &[name, var] : primitives[i].first->variables )
+		{
+			GeometricData::Interpretation interpret = IECore::getGeometricInterpretation( var.data.get() );
+
+			IECore::TypeId varTypeId = var.data->typeId();
+
+			if( var.interpolation == PrimitiveVariable::Constant )
+			{
+				varTypeId = vectorDataTypeFromDataType( var.data.get() );
+			}
+
+			PrimVarInfo &varInfo = varInfos.try_emplace( name, varTypeId, interpret, primitives.size() ).first->second;
+
+			if( varInfo.invalid )
+			{
+				continue;
+			}
+
+			if( varTypeId == IECore::InvalidTypeId )
+			{
+				msg( Msg::Warning, "mergePrimitives",
+					fmt::format(
+						"Discarding variable \"{}\" - Cannot promote Constant primitive variable of type \"{}\".",
+						std::string( name ), var.data->typeName()
+					)
+				);
+				varInfo.invalid = true;
+				continue;
+			}
+
+			if( varInfo.typeId != varTypeId )
+			{
+				msg( Msg::Warning, "mergePrimitives",
+					fmt::format(
+						"Discarding variable \"{}\" - types don't match: \"{}\" and \"{}\"",
+						name,
+						IECore::RunTimeTyped::typeNameFromTypeId( varInfo.typeId ),
+						IECore::RunTimeTyped::typeNameFromTypeId( var.data->typeId() )
+					)
+				);
+				varInfo.invalid = true;
+				continue;
+			}
+
+			if( !varInfo.interpretInvalid )
+			{
+				if( interpret != varInfo.interpret )
+				{
+					varInfo.interpret = GeometricData::Interpretation::Numeric;
```

Comment:
Looks like we deprecated `Numeric`, and this should be `None`.

Reply:
Addressed in https://github.com/GafferHQ/gaffer/commit/966690e2d0d3660edd266c6dff7c01baa81a4fb6